packages/code-du-travail-frontend/src/lib/useScrollLocation.js

```diff
@@ -12,40 +12,38 @@ function getWindowScrollPosition() {
   };
 }
 
-export function useWindowScrollPosition(effect, deps = []) {
-  const positionRef = useRef(getWindowScrollPosition());
+export function useWindowScrollPosition() {
+  const [scrollInfo, setScrollInfo] = useState({ direction: "down", y: 0 });
   const throttleMs = 30;
-  const throttleRef = useRef(null);
 
-  function triggerWindowPositionUpdate() {
-    const currentPosition = getWindowScrollPosition();
-    effect({
-      direction: currentPosition.y >= positionRef.current.y ? "down" : "up",
-      prevX: positionRef.current.x,
-      prevY: positionRef.current.y,
-      x: currentPosition.x,
-      y: currentPosition.y,
-    });
-    positionRef.current = currentPosition;
-    throttleRef.current = null;
-  }
-  const handleScroll = () => {
-    if (throttleMs) {
-      if (throttleRef.current === null) {
-        throttleRef.current = setTimeout(
-          triggerWindowPositionUpdate,
-          throttleMs
-        );
-      }
-    } else {
-      triggerWindowPositionUpdate();
+  const isThrottlingRef = useRef(false);
+  const previousPosition = useRef(getWindowScrollPosition());
+
+  const handleScroll = useCallback(() => {
+    if (!isThrottlingRef.current) {
+      isThrottlingRef.current = true;
+      setTimeout(() => {
+        const currentPosition = getWindowScrollPosition();
+        setScrollInfo({
+          direction:
+            currentPosition.y >= previousPosition.current.y ? "down" : "up",
+          prevX: previousPosition.current.x,
+          prevY: previousPosition.current.y,
+          x: currentPosition.x,
+          y: currentPosition.y,
+        });
+        isThrottlingRef.current = false;
+        previousPosition.current = currentPosition;
+      }, throttleMs);
     }
-  };
+  }, []);
 
-  useLayoutEffect(() => {
+  useEffect(() => {
```

Comment:
why did you change to useEffect ? useLayoutEffect is intentionnaly use here to get mesurement before render

Reply:
I have no idea why I did that, putting it back !

Reply:
Also I did a rapid test betwen branch and prod and it feels more laggy (tested on chromium 87.0.4263.0)

Reply:
But is there a way to have a scroll event before render ?